src/e4stypes/book_item.py

```diff
@@ -22,3 +31,24 @@ def get_edition(self) -> int:
 
     def get_course_number(self) -> int:
         return self._course_number
+
+    @classmethod
+    def add_item(cls, book: BookItem) -> BookItem:
+
+        # create book dict that will be added to the database
+        # id will be generated by mongo
+        book = {
+            'title': cls.title,
+            'description': cls.desc,
+            'price': cls.price,
+            'weight': cls.weight,
+            'seller': cls.seller,
+            'book_title': cls.book_title,
+            'edition': cls.edition,
+            'course_number': cls.course_number
```

Comment:
still not quite. Remember that `cls` refers to the class itself, not any instance of it. Since these are instance attributes, to access them we need a particular instance of that class (`book`).
```suggestion
            'title': book.get_title(),
            'description': book.get_desc(),
            'price': book.get_price(),
            'weight': book.get_weight(),
            'seller': book.get_seller(),
            'book_title': book.get_book_title(),
            'edition': book.get_edition(),
            'course_number': book.get_course_number()
```

same is true for the other types.

Reply:
Oh I see now! I originally tried doing something like `price: get_price()` and I was getting an error of sorts...I was missing the book class! Will fix these, thanks!